### DISCUSSION TITLE
use ISO-639-3 country code for inference language option for further merge of MMS models?

### DISCUSSION THREAD

**Discussion by ROBERT-MCDOWELL** [coqui-ai/TTS#4057] [Ideas]
Is it possible to make merge/link of xtts-v2 with the 1000+ MMS models from facebookt/mms-tts (https://huggingface.co/facebook/mms-tts)?

**Comment by eginhard** [comment]
I don't know what exactly you're asking:
1. Are the MMS models available in Coqui? Yes: https://github.com/coqui-ai/TTS?tab=readme-ov-file#example-text-to-speech-using-fairseq-models-in-1100-languages-
2. Can you use ISO-639-3 codes with other Coqui models? No. You'll need to make your own mapping if needed.
3. Can the MMS models be merged with XTTS? This doesn't make sense, they are completely different types of models.

**Comment by ROBERT-MCDOWELL** [comment]
1. thanks
2. ok
3. ok

**Comment by ROBERT-MCDOWELL** [comment]
among this function:
api = TTS("tts_models/deu/fairseq/vits")
api.tts_with_vc_to_file(
    "Wie sage ich auf Italienisch, dass ich dich liebe?",
    speaker_wav="target/speaker.wav",
    file_path="output.wav"
)

is it possible to load it from a local folder using the tts.load() like
            config.load_json(config_path)
            tts.load_checkpoint(config, checkpoint_dir=base_dir)

PS: is it better to follow more the forked https://github.com/idiap/coqui-ai-TTS   ?

**Reply by eginhard** [reply]
Yes, I recommend to use the fork - it has many bug fixes and works better with recent versions of dependencies.

In theory it's probably possible to load it like that, but because the Fairseq models are structured a bit differently you might have to manually call the function to convert it.

**Comment by ROBERT-MCDOWELL** [comment]
"you might have to manually call the function to convert it",
ha ok, any example please?

**Reply by eginhard** [reply]
Search for fairseq in the codebase and see how it's handled when the model name is specified, you'd have to do the same conversion when providing a path. Why is calling it by name not enough?

**Comment by ROBERT-MCDOWELL** [comment]
I'm developing an app that can be run offline, so the models must be loaded locally.
for now for xtts-v2 I use:

'''
                config = XttsConfig()
                config.load_json("/my/local/dir/config.json")
                tts = Xtts.init_from_config(config)
                tts.load_checkpoint(config, checkpoint_dir="/my/local/dir")
'''
if TTS("tts_models/deu/fairseq/vits") so can I grab what's downloaded to the cache and load it from another folder? if yes, I need to know how since I didn't find any example from fairseq to load the model locally. or maybe I completely misunderstood.

**Reply by eginhard** [reply]
Would you ship the models with the app or is it ok for the user to require an internet connection the first time they use the model? Because then you could keep loading the model by name, it will download it the first time it's used and then just loads the local file.

Doing it differently is possible, but you'll need to write the code for that yourself.

**Reply by ROBERT-MCDOWELL** [reply]
it's for completely offgrid with local low bandwidth network....
"Doing it differently is possible, but you'll need to write the code for that yourself"
ok but how if I don't have any doc nor example?
if I can load like TTS("tts_models/deu/fairseq/vits") so why not to use have the same principle of xtts-v2 loaded manually with config.json and checkpoint? sorry it's really not clear to me.

**Reply by eginhard** [reply]
As I wrote:
> Search for fairseq in the codebase and see how it's handled when the model name is specified, you'd have to do the same conversion when providing a path.

**Reply by ROBERT-MCDOWELL** [reply]
> As I wrote:
> 
> > Search for fairseq in the codebase and see how it's handled when the model name is specified, you'd have to do the same conversion when providing a path.

you mean at the fairseq git repo?

**Comment by ROBERT-MCDOWELL** [comment]
ok I understand a little more about the coqui-tts concept and models. thanks for your help!
